src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreRequest.cs

```diff
@@ -22,11 +23,21 @@ public class RestoreRequest
 
         private Lazy<LockFile> _lockFileLazy;
 
+        public RestoreRequest(
+             PackageSpec project,
+             RestoreCommandProviders dependencyProviders,
+             SourceCacheContext cacheContext,
+             ClientPolicyContext clientPolicyContext,
+             ILogger log) : this(project, dependencyProviders, cacheContext, clientPolicyContext, namespaces: null, log)
+        {
+        }
+
         public RestoreRequest(
             PackageSpec project,
             RestoreCommandProviders dependencyProviders,
             SourceCacheContext cacheContext,
             ClientPolicyContext clientPolicyContext,
+            PackageNamespacesConfiguration namespaces,
```

Comment:
Added a new parameter to the existing constructor. Added a new constructor with old signature for backward compatibility. Reference - https://github.com/NuGet/NuGet.Client/pull/4039/files#r645541058

Reply:
I think the reasoning there was the Download context in particular was used in many consumers of our library. 

Is that the case with RestoreRequest?

I would expect it's a scenario with a lot less traffic, so potentially this might not be an issue.

Reply:
So, it is okay to add a new parameter to the existing constructor and change all the existing references to pass `null for PackageNamespacesConfiguration`.  I will do that in the next commit.

Reply:
I reverted new constructor and added additional parameter to the existing constructor.